### PR TITLE
Include parameters in the data format 

### PR DESCRIPTION
**Pull request by TjarkMiener**
Following up on @LucaRomanato's PR #91.

Also some additional fields have been added to the format, which makes some transformations in transform.py redundant. This will give an overall speedup for the training process in CTLearn.

The reader will be updated in a separate PR.

### CODE REVIEW

**Comment by kosack** [comment]
What is the difference between this and ctapipe_io_magic (https://github.com/cta-observatory/ctapipe_io_magic)? If you are developing a ctapipe EventSource, it should be a separate package like that one, and then it becomes an automatically detected plugin.  After that, just using code like this will work generally for all formats supported (Sim-R0, LST-R1, DL1, MAGIC, etc.)

```py3
with EventSource(filename) as source:
    for event in source:
         process(event)
```
Will work no matter what format the file is in, as long as there is an  io plugin for it installed.  So if you separate this code which I would rename "MAGICDL1EventSource", if it generates data at the DL1 level), and either replace the old MagicEventSource (if it is not longer maintained), or make a new one, then you will save a lot of effort and it will be re-usable by everyone.

**Comment by TjarkMiener** [comment]
@kosack yes that's a very good suggestion! I think making it a separate ctapipe_io_MAGICDL1EventSource is a little bit overkill, since it only feature Deep Learning relevant information of the DL1 level. I will contacted the maintainers of ctapipe_io_magic and suggest to include it as a subclass of the main more general one. Let's see what they are thinking about this.

**Comment by TjarkMiener** [comment]
I contacted the MAGIC DL convenors about moving this EventSource as a subclass to the ctapipe_io_magic package. In the meantime, can this PR get merged? Anything missing for v0.9.0, @nietootein @rlopezcoto?

File: dl1_data_handler/writer.py [hunk 1]
```diff
@@ -465,52 +645,139 @@ def _create_event_table(self, subarray):
...
+                                                         "Parameter table of {} parameters, algorithm: {}, args: {}".format(tel_name,
+                                                                  self.cleaning_settings[index_parameters_table -1]['algorithm'],
+                                                                  self.cleaning_settings[index_parameters_table -1]['args']),
+                                                        filters=self.filters,
+                                                        expectedrows=expected_rows)
+
+                # Place blank image at index 0 of all image tables
+                parameter_row = parameter_table.row
+
+                parameter_row['event_index'] = -1
```
Comment (rlopezcoto) [inline dl1_data_handler/writer.py:742]: better to initialize values to `np.nan` or `None`, see the initialization of all these quantities in:
https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/io/lstcontainers.py

File: config/example_config.yml [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
+    image_extractor_settings:
+        algorithm: 'LocalPeakWindowSum'
+        args:
+            window_shift: 4
+            window_width: 8
+    # parameters0 table, cleaning_algo and cleaning_args
+    cleaning_settings:
+        algorithm: 'tailcuts_clean'
+        args:
+            picture_thresh: 7
```
[the remark this answers is not in the document]
Comment (TjarkMiener) [reply config/example_config.yml]: Solved

File: dl1_data_handler/writer.py (hunk 1 above)
Comment (TjarkMiener) [reply dl1_data_handler/writer.py:742]: I agree, it's better to use `np.nan` for initialization, but those are stored in the file, i.e. the data types have to match the with the real quantities. It's needed to construct the tables before.

Comment (TjarkMiener) [reply dl1_data_handler/writer.py:742]: Solved. Thanks for the hint @rlopezcoto :)

**Comment by TjarkMiener** [comment]
Hi @kosack! Following up on the discussion above, I found out that the ctapipe_io_magic is actually maintained in [this gitlab repo](https://gitlab.mpcdf.mpg.de/ievo/ctapipe_io_magic). I can ping the maintainers asking to synchronise the gitlab with the github repo of cta-observatory, so it could be directly detected by ctapipe as an external EventSource. So, the plan would be to leave the MAGIC related code here for the records and move to this EventSource once the transition of the ctapipe_io_magic to github is done and we adopt the stage1 for DL #92.

File: dl1_data_handler/writer.py (hunk 1 above)
Comment (maxnoe) [reply dl1_data_handler/writer.py:742]: That is an integer, -1 is the correct choice as there is no nan for integers and putting nan will result in the value being stored in the wrong data type.

**Review by rlopezcoto — APPROVED** [review]

**Review by nietootein — APPROVED** [review]

**Pull request merged**